### ISSUE TITLE
[Feature]: Authentication Page ( Sign Up )

### ISSUE THREAD

**Issue by Vin-Jex** [Vin-Jex/dent-clack#85]
### **Is your feature request related to a problem? Please describe.**

At the moment, the project does not have a dedicated signup page.

### **Describe the solution you'd like**

- Design and implement a **responsive signup page**, ensuring it integrates seamlessly with the existing layout used for the left content.
- Ensure the page adapts to both light and dark themes, maintaining a visually appealing design when toggling between themes.
- Include **loading states** for the buttons and **validation states** for the input fields to enhance user experience and feedback.

### **Describe alternatives you've considered**

I have considered using existing components from other authentication pages, but a dedicated, customized signup page will provide better control and a tailored user experience.

### **Additional context**

![Screenshot 2024-10-01 at 5 09 50 AM](https://github.com/user-attachments/assets/bf3324a5-8308-43aa-a8b5-1f8704e9f4f2)

**Comment by Ryadav0654** [comment]
Hi @Vin-Jex ,

I would like to work on this issue. Could you please assign it to me?

Thank you!

**Comment by Ryadav0654** [comment]
can i use react-hook-form?

**Comment by Vin-Jex** [comment]
> Hi @Vin-Jex ,
> 
> I would like to work on this issue. Could you please assign it to me?
> 
> Thank you!

Yes, I'll assign it to you, but you can't use react-hook-form since we're already developing our own custom inputs, buttons, and textareas. For now, feel free to use whatever works, but avoid installing any form-related packages until the custom reusable components are ready.

**Comment by Ryadav0654** [comment]
Hi @Vin-Jex ,
I have fixed all the issues and resolved the conflicts. A new pull request has been created. Please review it and let me know if any additional changes are required.
Thank you!
